pkg/asm/assignment_amd64_test.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+//go:build amd64 && linux
+// +build amd64,linux
+
+package asm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAssignment(t *testing.T) {
```

Comment:
It seems that `ParseAssignment` would be a perfect candidate for a little fuzzing tests, checking that at least basic fuzzing do not make it crash quickly https://go.dev/doc/security/fuzz/